types/src/transaction/mod.rs

```diff
@@ -2238,12 +2238,72 @@ impl TransactionListWithProof {
     }
 }
 
-/// This differs from TransactionListWithProof in that TransactionOutputs are
-/// stored (no transactions). Events are stored inside each TransactionOutput.
-///
-/// Note: the proof cannot verify the TransactionOutputs themselves. This
-/// requires speculative execution of each TransactionOutput to verify that the
-/// resulting state matches the expected state in the proof (for each version).
+// Note: this type must remain an enum to accommodate future changes/extensions
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub enum TransactionListWithProofV2 {
+    TransactionListWithAuxiliaryInfos(TransactionListWithAuxiliaryInfos),
+}
+
+impl TransactionListWithProofV2 {
+    pub fn new(transaction_list_with_auxiliary_infos: TransactionListWithAuxiliaryInfos) -> Self {
+        Self::TransactionListWithAuxiliaryInfos(transaction_list_with_auxiliary_infos)
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub struct TransactionListWithAuxiliaryInfos {
+    pub transaction_list_with_proof: TransactionListWithProof,
+    pub persisted_auxiliary_infos: Vec<PersistedAuxiliaryInfo>,
+}
+
+impl TransactionListWithAuxiliaryInfos {
+    pub fn new(
+        transaction_list_with_proof: TransactionListWithProof,
+        persisted_auxiliary_infos: Vec<PersistedAuxiliaryInfo>,
+    ) -> Self {
+        Self {
+            transaction_list_with_proof,
+            persisted_auxiliary_infos,
```

Comment:
nit: Could make an assertion that the lengths of transactions and auxilliary infos match.